tests/suite/scripting/destructuring.typ

```diff
@@ -373,3 +373,19 @@
 // Error: 6-12 too many elements to destructure
 // Hint: 6-12 the provided array has a length of 3, but the pattern expects 2 elements
 #for (x, y) in ((1,2,3), (4,5,6)) {}
+
+--- issue-4573-destructuring-unclosed-delimiter ---
+// See comment in issue for explanation. This one's complex.
```

Comment:
I think a short summary or fragment of your comment / the problem could be good. Something like "Ensure the parsing of the external assignment doesn't incorrectly lead the parser to an invalid state and to a subsequent panic when generating the unclosed delimiter error. See the comment in the issue for more explanation."